catalog/core/catalog-core-solr/src/main/java/ddf/catalog/source/solr/SolrMetacardClientImpl.java

```diff
@@ -367,9 +371,67 @@ protected SolrQuery postAdapt(
 
     setSortProperty(request, query, filterDelegate);
 
+    filterAttributes(request, query);
+
     return query;
   }
 
+  private void filterAttributes(QueryRequest request, SolrQuery query) {
+    if (skipFilteredAttributes(request)) {
+      return;
+    }
+
+    Set<String> excludedAttributes = (Set<String>) request.getPropertyValue(EXCLUDE_ATTRIBUTES);
+
+    Set<String> excludedFields =
+        resolver
+            .fieldsCache
+            .stream()
+            .filter(Objects::nonNull)
+            .filter(field -> excludedAttributes.stream().anyMatch(field::startsWith))
+            .collect(Collectors.toSet());
+
+    Set<String> wildcardFields =
+        SchemaFields.FORMAT_TO_SUFFIX_MAP
+            .values()
+            .stream()
+            .filter(suffix -> excludedFields.stream().noneMatch(field -> field.endsWith(suffix)))
+            .map(suffix -> "*" + suffix)
+            .collect(Collectors.toSet());
+
+    Set<String> includedFields =
+        SchemaFields.FORMAT_TO_SUFFIX_MAP
+            .values()
+            .stream()
+            .filter(suffix -> excludedFields.stream().anyMatch(field -> field.endsWith(suffix)))
+            .flatMap(
+                suffix ->
+                    resolver
+                        .fieldsCache
+                        .stream()
+                        .filter(Objects::nonNull)
+                        .filter(field -> field.endsWith(suffix))
+                        .filter(field -> excludedAttributes.stream().noneMatch(field::startsWith)))
+            .collect(Collectors.toSet());
+
+    Set<String> fields = Sets.union(includedFields, wildcardFields);
+
+    if (query.getFields() != null && query.getFields().length() > 2) {
+      fields = Sets.union(fields, Sets.newHashSet(query.getFields().substring(2).split(",")));
```

Comment:
❓ Should we worry about stripping spaces after splitting on `,`?

From their own code what I can tell is they [sometimes have spaces](https://github.com/apache/lucene-solr/blob/master/solr/solrj/src/java/org/apache/solr/client/solrj/SolrQuery.java#L1074) but then in other comments and code [don't have spaces](https://github.com/apache/lucene-solr/blob/master/solr/core/src/java/org/apache/solr/search/grouping/distributed/requestfactory/StoredFieldsShardRequestFactory.java#L70) and in the end looks like they split on [commas or spaces](https://github.com/apache/lucene-solr/blob/master/solr/core/src/java/org/apache/solr/handler/admin/LukeRequestHandler.java#L339) but don't filter out any empty values... 

So it _seems_ like nothing should happen even in odd cases (like somehow getting a double space) but it might be safest to just trim the fields anyways.